### PR TITLE
Updating AFE button, border, background, top color

### PR DESCRIPTION
**Pull request by hannahbergam**
This is a clean up PR focused on color changes for the AFE form to make it more closely match the spec and use our new branding colors. 

The continue button went from orange -> purple
The validation success color went from green -> teal
The border went from teal -> matching what Kelby has for the sidebar color
The background went from grey -> matching what Kelby has for the sidebar background
The teal dividers -> grey

I had to add another case for our ValidationStep component instead of changing the current 'SUCCESS' one because it's still used in three other places and I didn't want to move those to a new styling if the rest of their respecitve pages aren't updated yet.

| Before | After |
| ------ | ------ |
| Screenshot 2024-01-12 at 1 31 00 PM | Screenshot 2024-01-12 at 2 52 05 PM |

## Links

- spec: https://www.figma.com/file/qEFj4WmsjOC5okiMMxLJXq/AFE-Updates-(Dec-2023)?type=design&node-id=901-108&mode=design
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1368

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by kelbyhawn — COMMENTED** [review]
This looks great! 🙌 

One more update before it's good to go: can you make the teal divider lines between the sections light grey? You can use the `var(--neutral_dark20)` variable.

**Comment by hannahbergam** [comment]
> This looks great! 🙌
> 
> One more update before it's good to go: can you make the teal divider lines between the sections light grey? You can use the `var(--neutral_dark20)` variable.

I knew I was forgetting something!! Thank you!

**Review by kelbyhawn — APPROVED** [review]
Thanks for updating the lines! 😄

File: apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx [hunk 1]
```diff
@@ -401,11 +401,10 @@ const styles = {
     textAlign: 'right',
   },
   sectionBreak: {
-    borderColor: color.teal,
+    borderColor: 'var(--neutral_dark20)',
```
Comment (bethanyaconnor) [inline apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx]: @kelbyhawn, is there a difference between doing this versus using `color.neutral_dark20` (which pulls from `shared/css/color.scss`)?

Comment (kelbyhawn) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx]: Oh good point! I suggested that variable to match [these](https://github.com/code-dot-org/code-dot-org/blob/9c0d8b98a3dd6e9da0a6fc5c91b7730b6ecd372b/apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx#L326-L327), in the `amazonFutureEngineerEligibility.jsx` file, but the `color.neutral_dark20` one is probs the way to go.

Comment (kelbyhawn) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx]: On this note, is there documentation anywhere about when/where to use which color variable types in apps or dashboard?

Comment (bethanyaconnor) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx]: I don't think we have any documentation, other than "Extract colors into named, re-used variables." from https://github.com/code-dot-org/code-dot-org/blob/staging/STYLEGUIDE.md#css (which, frankly, isn't updated often). I personally prefer importing the `color.scss` file (which can be done in JS or in SCSS modules) and using those variables -- and I think that's the more common method used in our JS code -- but I don't think using the `var` syntax is necessarily wrong.

Comment (hannahbergam) [reply apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx]: Thanks for the documentation reference! Fwiw- I switched it:)

**Pull request merged**